src/main/java/com/xpanxion/java/springboot/da1/demo/service/student3/GymService3.java

```diff
@@ -27,5 +40,34 @@ public Gym3 addGym(Gym3 gym){
         return gymRepository.findByNameContaining(name);
     }
 
+    //
+    //Post
+    //
+
     public Member3 addMember(Member3 member) {return memberRepository.save(member);}
+
+    public CheckInOut3 addCheckIn(int memberId, String time, CheckType3 checkType3){
+        try {
+            Member3 member = memberRepository.findById(memberId).get();
+            CheckInOut3 checkInOut3 = new CheckInOut3(member, checkType3, time);
+            return checkInOutRepository.save(checkInOut3);
+        }catch (NoSuchElementException noSuchElementException) {
```

Comment:
Ah... and there it is: `NoSuchElementException` .  Here is another way to do it:

My Service has a method called `checkMember()`, to be interpreted as something like checking your baggage.

```
    public void checkMember(CheckType checkType, int memberId, Date time) {
        var member = getMemberOrThrow(memberId);
        var workoutHistory = new WorkoutHistory0(member, time, checkType);
        workoutHistoryRepository0.save(workoutHistory);
    }
```

```
    private Member0 getMemberOrThrow(int memberId) {
        var retval= memberRepository0.findById(memberId).orElseThrow(() ->
                new ResponseStatusException(HttpStatus.NOT_FOUND, "Member not found."));
        return retval;
    }
```

`OrElseThrow()` comes from Java Optionals.

I don't catch anything... I barf the exception out of the API and Postman displays it with the correct HTTP status code.